homeassistant/components/workday/binary_sensor.py

```diff
@@ -130,10 +132,16 @@ async def async_setup_entry(
 
     if province and province not in cls.subdivisions:
         LOGGER.error("There is no subdivision %s in country %s", province, country)
-        return
+        return None
 
-    obj_holidays = cls(
-        subdiv=province, years=year, language=cls.default_language
+    if language and language not in cls.supported_languages:
+        LOGGER.error("Language %s is not supported", language)
```

Comment:
We can probably make a resolvable issue here so the user can fix the issue.

Reply:
Sounds reasonable. Could you point towards an implementation example?

Reply:
Not used it myself yet but here is an implemented example https://github.com/home-assistant/core/blob/dev/homeassistant/components/unifiprotect/repairs.py#L88